tests/ifsbench/ifsbench_run.py

```diff
@@ -0,0 +1,366 @@
+    science_input = ecland_config.science[science]
+    tech_input = ecland_config.tech[tech]
+
+    if binary_path:
+        science_input.binary_path = Path(binary_path).resolve()
+
+    benchmark = build_ecland_benchmark(science=science_input, tech=tech_input)
+
+    job = science_input.job
+    if tasks:
+        job.tasks = tasks
+    if threads:
+        job.cpus_per_task = threads
+
+    arch = ecland_config.arch.get(arch, ecland_config.arch['default'])
+
+    if run_dir:
+        run_dir = Path(run_dir).resolve()
+        context = nullcontext(run_dir)
+    else:
+        context = TemporaryDirectory(dir=Path.cwd())
+
+    with context as run_dir:
+        run_dir = Path(run_dir)
+        benchmark.setup_rundir(run_dir)
+
+        bench_result = benchmark.run(
+            run_dir=run_dir,
+            job=job,
+            arch=arch,
+            launcher_flags=launcher_flags
+        )
+
+        result = EclandResult.from_rundir(
+            run_dir=run_dir,
+            stdout=bench_result.stdout,
+            stderr=bench_result.stderr,
+        )
+
+        with (run_dir/'result.yaml').open('w', encoding='utf-8') as f:
+            yaml.dump(result.dump_config(), f)
+
+        if validate_path:
+            validator = FrameCloseValidation(atol=0, rtol=0)
+            with Path(validate_path).open('r', encoding='utf-8') as f:
+                reference = EclandResult.from_config(yaml.safe_load(f))
+
+            if set(result.frames.keys()) != set(reference.frames.keys()):
+                raise RuntimeError("Results do not hold the same frames!")
+
+            for key in result.frames.keys():
+                frame = result.frames[key]
+                frame_ref = reference.frames[key]
+
+                equal, mismatch = validator.compare(frame, frame_ref)
+
+                if not equal and mismatch:
+                    idx, col = mismatch[0]
+                    error(f"First mismatch at ({idx}, {col}): {frame.loc[idx,col]} != {frame_ref.loc[idx,col]}.")
+
+                    for idx, col in mismatch:
+                        debug(f"Mismatch at ({idx}, {col}): {frame.loc[idx,col]} != {frame_ref.loc[idx,col]}.")
+
+
+@cli.command('validate', context_settings={"auto_envvar_prefix": "IFSBENCH"})
+@click.argument('result', type=click.Path(exists=True))
+@click.argument('reference', type=click.Path(exists=True))
+def validate(result, reference):
+    """
+    Compare two ecland result files and check for bit-identicality.
+    """
+    validator = FrameCloseValidation(atol=0, rtol=0)
+
+    with Path(result).open('r', encoding='utf-8') as f:
+        result = EclandResult.from_config(yaml.safe_load(f))
+
+    with Path(reference).open('r', encoding='utf-8') as f:
+        reference = EclandResult.from_config(yaml.safe_load(f))
+
+    if set(result.frames.keys()) != set(reference.frames.keys()):
+        raise RuntimeError("Results do not hold the same frames!")
+
+    for key in result.frames.keys():
```

Comment:
[no action] This loop looks fairly generic. I wonder if this could be internalised to IFSBench as a "default validation" or similar at some point? However, just a thought, does not need to be addressed now.